node-js-basics-101/node_modules/node-static/lib/node-static.js

```diff
@@ -0,0 +1,393 @@
+
+            /* Replace empty fields of differential requests by absolute values */
+            if (isNaN(byteRange.from) && !isNaN(byteRange.to)) {
+                byteRange.from = stat.size - byteRange.to;
+                byteRange.to = stat.size ? stat.size - 1 : 0;
+            } else if (!isNaN(byteRange.from) && isNaN(byteRange.to)) {
+                byteRange.to = stat.size ? stat.size - 1 : 0;
+            }
+
+            /* General byte range validation */
+            if (!isNaN(byteRange.from) && !!byteRange.to && 0 <= byteRange.from && byteRange.from < byteRange.to) {
+                byteRange.valid = true;
+            } else {
+                console.warn("Request contains invalid range header: ", rangeHeader);
+            }
+        } else {
+            console.warn("Request contains unsupported range header: ", rangeHeader);
+        }
+    }
+    return byteRange;
+}
+
+Server.prototype.respondNoGzip = function (pathname, status, contentType, _headers, files, stat, req, res, finish) {
+    var mtime           = Date.parse(stat.mtime),
+        key             = pathname || files[0],
+        headers         = {},
+        clientETag      = req.headers['if-none-match'],
+        clientMTime     = Date.parse(req.headers['if-modified-since']),
+        startByte       = 0,
+        length          = stat.size,
+        byteRange       = this.parseByteRange(req, stat);
+
+    /* Handle byte ranges */
+    if (files.length == 1 && byteRange.valid) {
+        if (byteRange.to < length) {
+
+            // Note: HTTP Range param is inclusive
+            startByte = byteRange.from;
+            length = byteRange.to - byteRange.from + 1;
+            status = 206;
+
+            // Set Content-Range response header (we advertise initial resource size on server here (stat.size))
+            headers['Content-Range'] = 'bytes ' + byteRange.from + '-' + byteRange.to + '/' + stat.size;
+
+        } else {
+            byteRange.valid = false;
+            console.warn("Range request exceeds file boundaries, goes until byte no", byteRange.to, "against file size of", length, "bytes");
+        }
+    }
+
+    /* In any case, check for unhandled byte range headers */
+    if (!byteRange.valid && req.headers['range']) {
+        console.error(new Error("Range request present but invalid, might serve whole file instead"));
+    }
+
+    // Copy default headers
+    for (var k in this.options.headers) {  headers[k] = this.options.headers[k] }
+    // Copy custom headers
+    for (var k in _headers) { headers[k] = _headers[k] }
+
+    headers['Etag']          = JSON.stringify([stat.ino, stat.size, mtime].join('-'));
+    headers['Date']          = new(Date)().toUTCString();
+    headers['Last-Modified'] = new(Date)(stat.mtime).toUTCString();
+    headers['Content-Type']   = contentType;
+    headers['Content-Length'] = length;
+
+    for (var k in _headers) { headers[k] = _headers[k] }
```

Comment:
__[jshint]__
* 'k' is already defined.

<!-- Comment by [SideCI](https://sideci.com) -->